### PR TITLE
Stencil library node from StencilFlow

### PR DESCRIPTION
**Pull request by definelicht**
Merge an adapted version of the StencilFlow stencil library node into the main repo for more general use.

### CODE REVIEW

**Review by alexnick83 — CHANGES_REQUESTED** [review]
Just rewrite the deprecated method call

File: dace/libraries/stencil/cpu.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            read_node = state.add_read(field)
+            input_dims = iterator_mapping[field]
+            input_shape = tuple(s for s, v in zip(shape, input_dims) if v)
+            data = sdfg.add_array(field, input_shape, dtype)
+            field_parameters = tuple(p for p, v in zip(parameters, input_dims)
+                                     if v)
+            for indices, connector in field_accesses[field].items():
+                access_str = ", ".join(
+                    f"{p} + ({i})" for p, i in zip(field_parameters, indices))
+                memlet = dace.Memlet.simple(field, access_str, num_accesses=-1)
```
[the remark this answers is not in the document]
Comment (definelicht) [reply dace/libraries/stencil/cpu.py]: Implemented in 279aae84146681868f5a465692563d3f8155dd07

File: dace/libraries/stencil/cpu.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+        # Add scalars as symbols
+        for field_name, mapping in iterator_mapping.items():
+            if not any(mapping):
+                sdfg.add_symbol(field_name, parent_sdfg.symbols[field_name])
+
+        #######################################################################
+
+        sdfg.parent = parent_state
+        sdfg._parent_sdfg = parent_sdfg  # TODO: this should not be necessary
```
[the remark this answers is not in the document]
Comment (definelicht) [reply dace/libraries/stencil/cpu.py]: Removed in 279aae84146681868f5a465692563d3f8155dd07

File: dace/libraries/stencil/stencil.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import dace
+import dace.library
+
+from .cpu import ExpandStencilCPU
+
+
+@dace.library.node
+class Stencil(dace.library.LibraryNode):
+    """
+    Represents applying a stencil that reads from constants offset from one or
```
Comment (alexnick83) [inline dace/libraries/stencil/stencil.py:14]:
```suggestion
    Represents applying a stencil that reads with constant offset from one or
```

**Review by alexnick83 — APPROVED** [review]
Minor typo(?)

File: dace/libraries/stencil/stencil.py (hunk 3 above)
Comment (definelicht) [reply dace/libraries/stencil/stencil.py]: Fixed in 3b5defbf

**Pull request merged**